streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -936,6 +1047,60 @@ private InternalTopicConfig createInternalTopicConfig(final StateStoreSupplier s
     }
 
     /**
+     * Get the Pattern to match all topics requiring to start reading from earliest available offset
+     * @return the Pattern for matching all topics reading from earliest offset, never null
+     */
+    public synchronized Pattern earliestResetTopicsPattern() {
```

Comment:
Does these need to be public APIs (note KStreamBuilder itself is a public class)? @mjsax has some ideas in refactoring the builder classes later to separate the user-facing APIs with those that are only used by internal topics, so for now maybe we just move them all to TopologyBuilder? WDYT @mjsax .